src/sentry/scripts/similarity/index.lua

```diff
@@ -278,507 +370,334 @@ local function scale_to_total(values)
     return result
 end
 
-local function collect_index_key_pairs(arguments, validator)
-    return build_variadic_argument_parser({
-        {"index", identity},
-        {"key", identity},
-    }, validator)(arguments)
+
+-- Signature Matching
+
+local function pack_frequency_coordinate(band, bucket)
+    return struct.pack('>B', band) .. bucket
```

Comment:
Unsure if assuming the band value is between [0, 255] is a good idea, but it sure makes packing and unpacking a lot easier.